src/Microsoft.Diagnostics.Runtime/src/DataTargets/ClrInfo.cs

```diff
@@ -26,122 +31,17 @@ public class ClrInfo : IComparable
-            if ((uint)val == 0x80070057)
-                return new LegacyRuntime(this, _dataTarget, lib, DesktopVersion.v4, 10000);
-
-            return new LegacyRuntime(this, _dataTarget, lib, DesktopVersion.v2, 3054);
-        }
-
-        /// <summary>
-        /// Creates a runtime from the given Dac file on disk.
-        /// </summary>
-        /// <param name="dacFilename">A full path to the matching mscordacwks for this process.</param>
-        /// <param name="ignoreMismatch">Whether or not to ignore mismatches between </param>
-        /// <returns></returns>
-        public ClrRuntime CreateRuntime(string dacFilename, bool ignoreMismatch = false)
-        {
-            if (string.IsNullOrEmpty(dacFilename))
-                throw new ArgumentNullException("dacFilename");
-
-            if (!File.Exists(dacFilename))
-                throw new FileNotFoundException(dacFilename);
-
-            if (!ignoreMismatch)
-            {
-                DataTarget.PlatformFunctions.GetFileVersion(dacFilename, out int major, out int minor, out int revision, out int patch);
-                if (major != Version.Major || minor != Version.Minor || revision != Version.Revision || patch != Version.Patch)
-                    throw new InvalidOperationException(string.Format("Mismatched dac. Version: {0}.{1}.{2}.{3}", major, minor, revision, patch));
-            }
-
-            return ConstructRuntime(dacFilename);
-        }
-
-#pragma warning disable 0618
-        private ClrRuntime ConstructRuntime(string dac)
-        {
-            if (IntPtr.Size != (int)_dataTarget.DataReader.GetPointerSize())
-                throw new InvalidOperationException("Mismatched architecture between this process and the dac.");
-
-            if (_dataTarget.IsMinidump)
-                _dataTarget.SymbolLocator.PrefetchBinary(ModuleInfo.FileName, (int)ModuleInfo.TimeStamp, (int)ModuleInfo.FileSize);
-
-            DacLibrary lib = new DacLibrary(_dataTarget, dac);
-
-            DesktopVersion ver;
-            if (Flavor == ClrFlavor.Core)
-            {
-                return new V45Runtime(this, _dataTarget, lib);
-            }
-
-            if (Version.Major == 2)
-            {
-                ver = DesktopVersion.v2;
-            }
-            else if (Version.Major == 4 && Version.Minor == 0 && Version.Patch < 10000)
-            {
-                ver = DesktopVersion.v4;
-            }
-            else
-            {
-                // Assume future versions will all work on the newest runtime version.
-                return new V45Runtime(this, _dataTarget, lib);
-            }
-
-            return new LegacyRuntime(this, _dataTarget, lib, ver, Version.Patch);
-        }
+        public string DacFileName => ClrInfoProvider.GetDacFileName(Flavor, IsLinux);
+        public string DacRequestFileName => ClrInfoProvider.GetDacRequestFileName(Flavor, Architecture, Architecture, Version, IsLinux);
```

Comment:
I see now why you want IsLinux, but you should go through DataTarget.GetPlatform() == Linux (or something similar).